sources/Inputs/src/Configuration.cpp

```diff
@@ -64,6 +64,21 @@ updateActivePowerCompensationValue(Configuration::ActivePowerCompensation& activ
     LOG(warn) << MESS(BadActivePowerCompensation, apcString) << LOG_ENDL;
   }
 }
+
+/**
+ * @brief Helper function to update an a std::chrono::seconds value
+ *
+ * @param seconds the value to update
+ * @param tree the element of the boost tree
+ * @param key the key of the parameter to retrieve
+ */
+static void
+updateSeconds(std::chrono::seconds& seconds, const boost::property_tree::ptree& tree, const std::string& key) {
+  auto value_opt = tree.get_child_optional(key);
```

Comment:
duplcate code: should called updateValue<unsigned int> like updateActivePowerCompensationValue

Reply:
Update to call updateValue. But we have to make the update only if the key exists. If we always update the final seconds variable:
```
unsigned int scount;
helper::updateValue(scount, tree, key);
seconds = std::chrono::seconds(scount);
```
Then default values could be overwritten when the key is not present in the config file. So we have to write it like this:
```
if (tree.get_child_optional(key)) {
  unsigned int scount;
  helper::updateValue(scount, tree, key);
  seconds = std::chrono::seconds(scount);
}
```
That forces a double check for key existence, one in the `updateSeconds` and another one in `updateValue`. 

As an alternative, the value could be read as string, but then a conversion would be needed, so we think this is more clear.

Reply:
Put a default value to the integer before calling updateValue (a value that is not possible like infinity or something) and then check if the value has been updated by the call of updateValue, just like it is done in updateActivePowerCompensationValue

Reply:
Done. I have used `MAX_INT` as a marker for value not updated.

Reply:
Done. Modified using `numeric_limits<unsigned int>::max()` to mark invalid value.

Reply:
Modified to avoid using intermediate variable.